fix isRoofProbablyVisibleFromBelow

Helium314 opened the pull request:
> The app should probably interpret a missing roof:levels as roof:levels=0 for the roof shape quest.

fix #2666 
Additionally now the quite common fractional value of `roof:levels` (often 0.5 or 1.5) is now understood correctly. (according to [this](https://play.kotlinlang.org/#eyJ2ZXJzaW9uIjoiMS40LjMwIiwicGxhdGZvcm0iOiJqYXZhIiwiYXJncyI6IiIsImpzQ29kZSI6IiIsIm5vbmVNYXJrZXJzIjp0cnVlLCJ0aGVtZSI6ImlkZWEiLCJjb2RlIjoiLyoqXG4gKiBZb3UgY2FuIGVkaXQsIHJ1biwgYW5kIHNoYXJlIHRoaXMgY29kZS4gXG4gKiBwbGF5LmtvdGxpbmxhbmcub3JnIFxuICovXG5cbmZ1biBtYWluKCkge1xuICAgIHZhbCByb29mTGV2ZWxzID0gXCIxLjVcIlxuICAgIHByaW50bG4ocm9vZkxldmVscz8udG9GbG9hdE9yTnVsbCgpID86IDApXG4gICAgcHJpbnRsbihyb29mTGV2ZWxzPy50b0ludE9yTnVsbCgpID86IDApXG59In0=))

westnordost commented:
Hey, thank you! But it would be required to also add a test that confirms that it works

westnordost commented:
(While being in that class, press Ctrl+Shift+T to directly jump to the test class)

westnordost commented:
If you are not up to it, let me know. I will do that then

Helium314 commented:
I added some tests and switched to using float for all levels for consistency, even though it hardly matters in the case if `building:levels`
I hope the tests are acceptable:
- included high buildings with no or 0 `roof:levels`
- added test for high buildings with many `roof:levels`
- changed test for 0 or no `roof:levels` to be unknown only for buildings with no or few levels
- included fractional `roof:levels` and `building:levels`

westnordost commented:
Looks good, thank you!
